app/controllers/preauthorize_transactions_controller.rb

```diff
@@ -146,6 +159,52 @@ def validate_booking(tx_params:, quantity_selector:)
       end
     end
 
+    def all_days_available(timeslots, start_on, end_on)
+      # Take all the days except the exclusive last day
+      requested_days = (start_on..end_on).to_a[0..-2]
+
+      available_days = timeslots
+                         .select { |s| s[:unitType] == :day }
+                         .map { |s| s[:start].to_date }
+
+      requested_days.all? { |d|
```

Comment:
This is perfectly fine, no need to change, but If you want to make this without iterating, you could turn these arrays to Sets and use set operations.